lib/crawler.js

```diff
@@ -230,42 +237,222 @@ util.inherits(Crawler, EventEmitter);
 Crawler.prototype.start = function() {
     var crawler = this;
 
-    // only if we haven't already got stuff in our queue...
-    crawler.queue.getLength(function(err, length) {
-        if (err) {
-            throw err;
-        }
+    crawler.getRobotsTxt(function (error, robotsTxtUrl, robotsTxtBody) {
+        if (!error && crawler.respectRobotsTxt) {
+            var robots = robotsTxtParser(robotsTxtUrl, robotsTxtBody);
 
-        if (!length) {
-
-            // Initialise our queue by pushing the initial request data into it...
-            crawler.queue.add(
-                crawler.initialProtocol,
-                crawler.host,
-                crawler.initialPort,
-                crawler.initialPath,
-                QUEUE_ITEM_INITIAL_DEPTH,
-                function(error) {
-                    if (error) {
-                        throw error;
-                    }
-                });
+            crawler._robotsTxtFetchCondition = function (parsedURL) {
+                parsedURL.hostname = parsedURL.host;
+
+                var formattedURL = uri(parsedURL).href(),
+                    allowed = false;
+
+                // The punycode module sometimes chokes on really weird domain
+                // names. Catching those errors to prevent crawler from crashing
+                try {
+                    allowed = robots.isAllowed(formattedURL, crawler.userAgent);
```

Comment:
Do we want to emit a `fetcherror` event or similar? I figure we probably shouldn't silently swallow the URLs.